controllers/chat/getSignedChannelList.js

```diff
@@ -14,22 +22,34 @@ const __queryChannelBuilder = async (userId, limit, offset) => {
   return channels;
 };
 
-const __transformChannelData = (channel) => {
+/**
+ *
+ * @param {import('stream-chat').Channel} channel
+ * @returns
+ */
+const __filterAndTransformChannelData = (acc, channel) => {
   const newChannel = {...channel.data};
   delete newChannel.config;
   delete newChannel.own_capabilities;
 
+  const messageLength = channel.state.messages?.length;
+  const channelType = newChannel?.channel_type;
+
+  if (messageLength === 0 && channelType === CHANNEL_TYPE.TOPIC) return acc;
+
   const members = [];
   Object.keys(channel.state.members).forEach((member) => {
     members.push(channel?.state?.members[member]);
   });
 
-  return {
+  acc.push({
     ...newChannel,
     members,
     unreadCount: channel.state.unreadCount,
     messages: channel.state.messages
-  };
+  });
+
+  return acc;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new function `__filterAndTransformChannelData` is a good improvement over the old `__transformChannelData` function. It now filters out channels with zero messages and of a specific type (`CHANNEL_TYPE.TOPIC`). However, it's important to ensure that this change doesn't break any existing functionality that relies on these channels being present in the returned list.



<!-- This is an auto-generated comment by OSS CodeRabbit -->